lms/djangoapps/bookmarks/services.py

```diff
@@ -43,13 +77,20 @@ def is_bookmarked(self, usage_key):
         Returns:
             Bool
         """
-        try:
-            api.get_bookmark(user=self._user, usage_key=usage_key)
-        except ObjectDoesNotExist:
-            log.error(u'Bookmark with usage_id: %s does not exist.', usage_key)
+        course_key = modulestore().fill_in_run(usage_key.course_key)
+        if course_key.run is None:
             return False
+        cache_key = CACHE_KEY_TEMPLATE.format(self._user.id, course_key)
+        request_cache = RequestCache.get_request_cache()
 
-        return True
+        if not cache_key in request_cache.data:
+            self.bookmarks(course_key)
+
+        for bookmark in request_cache.data.get(cache_key, []):
+            if bookmark['usage_id'] == unicode(usage_key):
```

Comment:
Nit: you're going to serialize usage key once per bookmarked item, rather than just once.